src/hooks/control.ts

```diff
@@ -84,33 +81,46 @@ const getDataOptions = async (userControlGridObject: UserControlGridProps) => {
  * @param configData Selector configuration, template params and setDataOptions state method
  * @returns modified configData.templateParams
  */
-const setControlData = (configData: UserControlGridProps) => {
-  configData?.userControlConfig?.map((currentConfig: UserControlConfig) => {
-    const paramKey = currentConfig?.url_param_key;
-    const uid = currentConfig?.uid;
-    const valueKey = currentConfig?.request_info?.value_key;
-    const defaultValue = currentConfig?.request_info?.default_value;
-    configData.templateParams.$control_values = {
-      ...configData.templateParams.$control_values,
-      [uid]: {
-        values: {},
-      },
-    };
-    if (paramKey) {
-      const paramValue = getQueryParam(windowRef.location.href, paramKey);
-      if (paramValue) {
-        configData.templateParams.$control_values[uid].values = {
-          [valueKey]: paramValue //use url_param value
-        };
-      }
-    }
-    if (defaultValue) {
-      configData.templateParams.$control_values[uid].values = {
-        [valueKey]: defaultValue //else use default value
-      };
-    }
-  })
-  return configData.templateParams;
+export const setControlData = (configData: UserControlConfig[], setTemplateParams: any) => {
```

Comment:
I'm unsure why this hook has to be separate from the `useControl` hook. 

We should be collecting the data for the useControl hook before we initialize it instead of initializing it before data is ready and then updating the data after.